File: domain/src/main/java/com/example/movie/entity/movie/Movie.java
package com.example.movie.entity.movie;

import com.example.BaseEntity;
import jakarta.persistence.*;

import java.time.LocalDate;

@Entity
public class Movie extends BaseEntity {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private String thumbnail;

    @Enumerated(EnumType.STRING)
    private Genre genre;

    @Enumerated(EnumType.STRING)
    private Grade grade;

    private LocalDate releaseDate;

    private Long runningTime;

}

File: movies/src/main/kotlin/com/redis/domain/Movie.kt
package com.redis.movies.domain

import com.redis.utils.BaseEntity
import jakarta.persistence.*
import java.time.LocalDateTime

@Table(name = "movie")
@Entity
class Movie(
    @Column(name = "title")
    val title: String,

    @Column(name = "film_ratings")
    @Enumerated(EnumType.STRING)
    val filmRatings: FilmRatings,

    @Column(name = "release_date")
    val releaseDate: LocalDateTime,

    @Column(name = "thumbnail_image_path")
    val thumbnailImagePath: String,

    @Column(name = "running_time")
    val runningTime: Long,

    @OneToOne(mappedBy = "movie", cascade = [CascadeType.ALL], orphanRemoval = true)
    val genre: MovieGenre,

    @Column(name = "cinema_id")
    val cinemaId: Long,

    @Embedded
    val screeningSchedules: ScreeningSchedules = ScreeningSchedules(),

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "movie_id")
    val id: Long? = null
): BaseEntity() {

}